OVERVIEW.md

```diff
@@ -0,0 +1,66 @@
+# Overview #
+
+Here it is described a brief overview of xdist's internal architecture.
+
+
+`xdist` works by spawning one or more **worker nodes**, which are controlled
+by the **master node**. Each **worker node** is responsible for performing 
+a full test collection and afterwards running tests as dictated by the **master node**.
+
+The execution flow is:
+
+1. **master node** spawns one or more **worker nodes** at the begginning of
+   the test session. The communication between **master** and **worker** nodes makes use of 
+   [execnet](http://codespeak.net/execnet/) and its [gateways](http://codespeak.net/execnet/basics.html#gateways-bootstrapping-python-interpreters).
+   The actual interpreters executing the code for the **worker nodes** might
+   be remote or local. 
+  
+1. Each **worker node** itself is a mini pytest runner. **workers** at this
+   point perform a full test collection, sending back the collected 
+   test-ids back to the **master node** which does not
+   perform any collection itself.
```

Comment:
If the collection was performed by master then it would have to serialize collected items to send them through the wire, as workers live in another process. The problem is that test items are not easily (impossible?) to serialize, as they contain references to the test functions, fixture managers, config objects, etc. Even if one manages to serialize it, it seems it would be very hard to get it right and easy to break by any small change in pytest. At least that's my understanding.

IIRC @hpk42 commented in a mailing list that `xdist` at its beginning performed collection at **master**, but that proved too troublesome. Perhaps he can share more here.